core/state/statedb.go

```diff
@@ -320,6 +341,7 @@ func (self *StateDB) Copy() *StateDB {
 	// ignore error - we assume state-to-be-copied always exists
 	state, _ := New(common.Hash{}, self.db)
 	state.trie = self.trie
+	state.all = self.all
```

Comment:
All contains only the non-modified state objects (i.e. whatever's in the db currently). It's updated only when the statedb commits.